device/core/src/sak_authentication_provider.ts

```diff
@@ -45,7 +47,14 @@ export class SharedAccessKeyAuthenticationProvider extends EventEmitter implemen
     }
 
     /*Codes_SRS_NODE_SAK_AUTH_PROVIDER_16_002: [The `constructor` shall start a timer that will automatically renew the token every (`tokenValidTimeInSeconds` - `tokenRenewalMarginInSeconds`) seconds if specified, or 45 minutes by default.]*/
-    this._renewToken();
+
+    if (startTimerOnNextTick) {
+      // we kick off the initial token creation during the next tick because if this code is being run from a sub-class then
+      // there might be additional initialization that needs to occur before we are ready to sign tokens
+      setImmediate(() => this._renewToken());
+    } else {
+      this._renewToken();
+    }
```

Comment:
that feels slightly sketchy. can you not document this parameter so that we revisit this later without it being a breaking change?

Reply:
Yes, I am not super happy about this. I'll remove the documentation. Also, if you have another idea I am happy to change this.